R/tmiic.plot.R

```diff
@@ -0,0 +1,444 @@
+    stop("Method not supported")
+  }
+  return(tmiic.getIgraph(tmiic.res))
+}
+
+#-----------------------------------------------------------------------------
+# tmiic.getIgraph
+#-----------------------------------------------------------------------------
+#' Igraph plotting function for tmiic (temporal mode of miic)
+#'
+#' @description This functions returns an igraph object built from the result
+#' returned by:
+#' \code{\link{miic}} executed in temporal temporal mode.
+#' \code{\link{tmiic.flatten_network}}
+#' 
+#' @details
+#' Edges attributes are passed to the igraph graph and can be accessed with
+#' e.g. \code{E(g)$partial_correlation}. See \code{\link{miic}} for more
+#' details on edge parameters. By default, edges are colored according to the
+#' partial correlation between two nodes conditioned on the conditioning set
+#' (negative is blue, null is gray and positive is red) and their width is
+#' based on the conditional mutual information minus the complexity cost.
+#'
+#' @param tmiic.res [a tmiic graph object]
+#' The graph object returned by the miic execution in temporal mode, 
+#' eventually flattened
+#'
+#' @return An igraph graph object.
+#' 
+#' @export
+#'
+#' @seealso
+#' \code{\link{miic}} for details on edge parameters in the returned object,
+#' \code{\link[igraph]{igraph.plotting}} for the detailed description of the
+#' plotting parameters and \code{\link[igraph]{layout}} for different layouts.
+#-----------------------------------------------------------------------------
+tmiic.getIgraph <- function (tmiic.res){
+  if (class(tmiic.res) != "tmiic") {
+    stop("Not a tmiic object.")
+  }
+  
+  class(tmiic.res) <- "miic"
+  graph <- getIgraph(tmiic.res)
+  class(tmiic.res) <- "tmiic"
+  
+  if ( tmiic.isLaggeg (colnames(tmiic.res$adj_matrix)) ) {
+    igraph::V(graph)$label.dist = 1
+    igraph::V(graph)$label.degree = pi/2
+    igraph::E(graph)$curved = TRUE
+  }
+  else {
+    igraph::E(graph)$curved = FALSE
+    igraph::E(graph)$label <- tmiic.res$all.edges.summary$lag
+  }
+  return(graph)
+}
+ 
+#-----------------------------------------------------------------------------
+# tmiic.isLaggeg
+#-----------------------------------------------------------------------------
+# Test if the list of nodes corresponds to a lagged or non lagged graph
+#
+# @param list_nodes [a list of string]
+#
+# @return A boolean value, TRUE if the network is lagged, FALSE otherwise
+#-----------------------------------------------------------------------------
+tmiic.isLaggeg <- function (list_nodes) {
+  is_graph_lagged = TRUE
+  for (one_node in list_nodes) {
+    regex_found = grepl ("_lag.*$", one_node)
+    if (!regex_found) {
+      is_graph_lagged = FALSE
+      break
+    }
+  }
+  return (is_graph_lagged)
+}
+
+#-----------------------------------------------------------------------------
+# tmiic.getNbNodesNonLagged
+#-----------------------------------------------------------------------------
```

Comment:
These counting is called from the plot part (where we do not have the non lagged input data anymore) and avoid to the user to specify an extra parameter not needed as we can determine ourselves the number of nodes not lagged.

Reply:
This extra parameter is not for user to specify, a parameter can be internal, just as is an internal function. All those lagged nodes are created by tmiic, so it's natural for it (through the `tmiic` object for example) to remember what it has done. Instead of keeping the result directly, you are using a function here to help it recall what it did in the past, which is imho unnecessary complication.

Reply:
It can be done in such way and the same for isLaggeg. For me it is OK if you prefer to store these info in the tmiic object.

Reply:
I added in the "tmiic_specific" list of the object returned by miic a "n_nodes_not_lagged" integer. So the function will be dropped in the next push